core/src/main/scala/org/apache/spark/rdd/HadoopRDD.scala

```diff
@@ -219,6 +220,9 @@ class HadoopRDD[K, V](
       val bytesReadCallback = if (split.inputSplit.value.isInstanceOf[FileSplit]) {
         SparkHadoopUtil.get.getFSBytesReadOnThreadCallback(
           split.inputSplit.value.asInstanceOf[FileSplit].getPath, jobConf)
+      } else if (split.inputSplit.value.isInstanceOf[CombineFileSplit]) {
+        SparkHadoopUtil.get.getFSBytesReadOnThreadCallback(
+          split.inputSplit.value.asInstanceOf[CombineFileSplit].getPath(0), jobConf)
```

Comment:
Yes, SparkHadoopUtil) can check for those classes. It can have a matcher on the 4 classes (2 new and 2 old). So the call from hadoopRdd would be something like:
SparkHadoopUtil.get.getFSBytesReadOnThreadCallback(split.inputSplit, jobConf)
Not a big deal i guess since in SparkHadoopUtil you'll have four cases but at least that logic is centralized.